src/UI/NavigationContainer.elm

```diff
@@ -168,31 +318,92 @@ stateUpdate msg state =
 -- Constructors
 
 
+{-| The correct way of instantiating a [`Nav.State`](UI-NavigationContainer#State).
+
+    Nav.stateInit renderConfig
+
+-}
 stateInit : RenderConfig -> State
 stateInit cfg =
     { menuExpanded = not <| RenderConfig.isMobile cfg }
 
 
+{-| `Nav.contentSingle` indicates that the current page is a simple single page.
+It expects the final page's view in the only parameter.
+
+    Nav.contentSingle <| view renderConfig model
+
+-}
 contentSingle : Element msg -> Content msg
 contentSingle body =
     ContentSingle body
 
 
+{-| `Nav.contentStackChild` indicates that the current page is a stack child's page.
+It expects the child's configuration and the final page's view as its parameters.
+
+    Nav.contentStackChild
+        { title = "Edit: Card " ++ selectedCard.number
+        , buttons =
+            [ Icon.print "Print card"
+                |> Button.fromIcon
+                |> Button.cmd (Msg.PrintCard selectedCard) Button.light
+            ]
+        , goBackMsg = Msg.DiscardCardChanges
+        }
+    <|
+        cardEditView renderConfig selectedCard
+
+-}
 contentStackChild : StackChild msg -> Element msg -> Content msg
 contentStackChild prop body =
     ContentStackChild prop body
 
 
+{-| `Nav.menuPage` describes a page to [`Nav.withMenuPages`](UI-NavigationContainer#withMenuPages).
+
+    Nav.menuPage (Icon.edit "Edit cards")
+        (Link.link "/edit-cards")
+        (currentPage == Pages.CardsEdit)
+
+-}
 menuPage : Icon -> Link -> Bool -> MenuPage
 menuPage icon link isCurrent =
     Menu.Page icon link isCurrent
 
 
+{-| `Nav.menuPage` describes an action to [`Nav.withMenuActions`](UI-NavigationContainer#withMenuActions).
+
+    Nav.menuAction
+        (Icon.logout "Logout")
+        Msg.SessionLogout
+
+-}
 menuAction : Icon -> msg -> MenuAction msg
 menuAction icon msg =
     Menu.Action icon msg
 
 
+{-| `Nav.navigator` holds the minimum amount of information required for all the features (menu, dialogs, and page's layout) to work.
+
+The first parameter is a lambda that should transform [`Nav.Msg`](UI-NavigationContainer#Msg) in a message type controlled by the user.
```

Comment:
I don't think `lambda` is a very used term in the Elm community, maybe I would change to function